subsys/net/lib/lwm2m/lwm2m_engine.c

```diff
@@ -575,6 +575,34 @@ static int engine_remove_observer(const uint8_t *token, uint8_t tkl)
 	return 0;
 }
 
+static int engine_remove_observer_by_path(struct lwm2m_obj_path *path)
+{
+	struct observe_node *obs, *found_obj = NULL;
+	sys_snode_t *prev_node = NULL;
+
+	/* find the node index */
+	SYS_SLIST_FOR_EACH_CONTAINER(&engine_observer_list, obs, node) {
+		if (memcmp(path, &obs->path, sizeof(*path)) == 0) {
+			found_obj = obs;
+			break;
+		}
+
+		prev_node = &obs->node;
+	}
+
+	if (!found_obj) {
+		return -ENOENT;
+	}
+
+	LOG_INF("Removing observer for path %u/%u/%u/%u",
```

Comment:
This would suggest that Resource instance was always cancelled (which isn't even a valid observation), at least the path level should be included in the log, or even better some smart formatting applied (to print the actual path cancelled).

Reply:
sure, good feedback